### PR TITLE
Encrypt messages

### PR DESCRIPTION
**Pull request by doomspork**

### CODE REVIEW

**Comment by dmvt** [comment]
I like it, but I think the lack of a configured encryption key should bypass the encryption layer so it can still be run unencrypted in dev mode

File: lib/exque/consuming/consumer.ex [hunk 1]
```diff
@@ -19,28 +19,32 @@ defmodule Exque.Consuming.Consumer do
...
-            {:ok, res} -> {:ok, res}
-            :ok -> {:ok, nil}
-            res -> {:ok, res}
-          end
-        rescue
-          e -> {:error, e}
-        end
+        action
+        |> perform_action(message)
+        |> report_result(channel, tag, message)
```
Comment (joshleecreates) [inline lib/exque/consuming/consumer.ex]: This is great

File: lib/exque/consuming/router.ex [hunk 2]
```diff
@@ -39,15 +39,17 @@ defmodule Exque.Consuming.Router do
       end)
     end
 
-    defp define_route(topic, consumer, mapping) when is_list(consumer) do
-      define_route(topic, consumer |> List.last, mapping)
-    end
-
+    defp define_route(topic, consumer, mapping) when is_list(consumer),
+      do: define_route(topic, List.last(consumer), mapping)
```
Comment (joshleecreates) [inline lib/exque/consuming/router.ex]: LOL at my original use of a pipe here. Thanks.

File: lib/exque/consuming/router.ex [hunk 3]
```diff
@@ -232,25 +239,46 @@ defmodule Exque.Consuming.Router do
...
             def handle_info(:reconnect, state) do
               request_channel(state)
               {:noreply, state}
             end
 
-            def terminate(reason, _) do
-              Logger.debug("Termination detected because #{inspect reason}")
-            end
+            def terminate(reason, _),
+              do: Logger.debug("Termination detected because #{inspect reason}")
```
Comment (joshleecreates) [inline lib/exque/consuming/router.ex]: Should this be a debug? I'd think we'd want this in production.

Comment (doomspork) [inline lib/exque/consuming/router.ex]: That's a good point.  This should either be a warn or an error.  I'll wait to see what @dmvt thinks.

Comment (doomspork) [inline lib/exque/consuming/router.ex]: We use warn elsewhere so I'll change this.

File: lib/exque/consuming/router.ex [hunk 4]
```diff
@@ -316,8 +336,8 @@ defmodule Exque.Consuming.Router do
       end
 
       defp spawn_channel(conn, topic, consumer) do
-        #TODO: consider adding configuration to change exque prefix
-        queue = "exque.#{Exque.Utils.app(__MODULE__) |> Macro.underscore}.#{topic}"
+        prefix = Application.get_env(:exque, :prefix, :exque)
```
Comment (doomspork) [inline lib/exque/consuming/router.ex]: @dmvt whatcha think?  There was a TODO here to handle the prefix, this does the trick.

Comment (joshleecreates) [inline lib/exque/consuming/router.ex]: 👍

**Pull request merged**